### DISCUSSION TITLE
Get-TypeAccelerator output

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [SamErde/PSPreworkout#20] [General]
This command has the same challenges as others in the module we've discussed. I know the module started out based on things that made *your* life easier. But now that you are creating a public module, I think you need to re-assess things. `Get-TypeAccelerator` has a `-GridView` parameter. This breaks the paradigm of having functions do one thing. Your function should get the information. Let the user decide how to format or present it.

In this case, if I'm in PowerShell 7, `-GridView` uses `Out-ConsoleGridview`. If I wanted to use `Out-Gridview` I have to run `Get-TypeAccelerator | Out-GridView`.  In which case that could just as easily be `Get-TypeAccelerator | Out-ConsoleGridview`.  You're approach also takes away my ability to use `Out-ConsoleGridView` parameters.

**Comment by SamErde** [comment]
I had to read it twice, but yes, I get what you are saying now. 💡 🤌

**Comment by SamErde** [comment]
Removed the **GridView** parameter and code in #29.

Lessons learned:

- Have functions do one thing. (Don't combine GET and SHOW.)
- Don't get in the way of the user's own potential preferences or environmental limitations.

**Reply by jdhitsolutions** [reply]
And see how this dramatically changes how you code.
